src/ansys/geometry/core/geometry/surfaces/plane.py

```diff
@@ -0,0 +1,230 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Provides for creating and managing a cylinder."""
+from functools import cached_property
+
+from beartype import beartype as check_input_types
+from beartype.typing import Union
+import numpy as np
+
+from ansys.geometry.core.geometry.parameterization import (
+    Interval,
+    Parameterization,
+    ParamForm,
+    ParamType,
+    ParamUV,
+)
+from ansys.geometry.core.geometry.surfaces.surface import Surface
+from ansys.geometry.core.geometry.surfaces.surface_evaluation import SurfaceEvaluation
+from ansys.geometry.core.math import (
+    UNITVECTOR3D_X,
+    UNITVECTOR3D_Z,
+    Matrix44,
+    Point3D,
+    UnitVector3D,
+    Vector3D,
+)
+from ansys.geometry.core.typing import Real, RealSequence
+
+
+class PlaneSurface(Surface):
+    """
+    Provides 3D plane surface representation.
+
+    Parameters
+    ----------
+    origin : Union[~numpy.ndarray, RealSequence, Point3D],
+        Centered origin of the torus.
+    reference : Union[~numpy.ndarray, RealSequence, UnitVector3D, Vector3D]
+        X-axis direction.
```

Comment:
Should the description for this parameter and the next be the same? I assume not.